### PR TITLE
Added instant recovery retention

### PR DESCRIPTION
**Pull request by nilaydshah**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

* Added instant recovery retention

### CODE REVIEW

File: 101-recovery-services-weekly-backup-policy-create/azuredeploy.json [hunk 1]
```diff
@@ -32,6 +32,14 @@
         "description": "Any Valid timezone, for example:UTC, Pacific Standard Time. Refer: https://msdn.microsoft.com/en-us/library/gg154758.aspx"
       }
     },
+    "instantRpRetentionRangeInDays": {
+      "type": "int",
+      "defaultValue": 5,
+      "allowedValues": [5],
```
Comment (bmoore-msft) [inline 101-recovery-services-weekly-backup-policy-create/azuredeploy.json]: why is this a param if it only allows on value?

Comment (nilaydshah) [reply 101-recovery-services-weekly-backup-policy-create/azuredeploy.json]: Fixed this. Removed this parameter and passed directly value

**Comment by bmoore-msft** [comment]
@nilaydshah - we're going to have to put these on hold.  The resources created by these templates (recovery services vaults) cannot be deleted - so I don't want to merge until that gets resolved.

**Comment by nilaydshah** [comment]
Vaults are deletable until it has not protected any backup item. And unless you remove those protected item first, as security reasons, we don't allow deletion of vaults. But for this template, we are just creating vault and backup policy, which are deletable, as it's not protecting VM or Workload. Can I know what error we are getting while deletion to debug further?

**Comment by bmoore-msft** [comment]
Send me an email and I'll send you the ICM that has details.

**Pull request merged**